cpp/ql/src/semmle/code/cpp/models/implementations/Swap.qll

```diff
@@ -15,3 +18,27 @@ private class Swap extends DataFlowFunction {
     output.isParameterDeref(0)
   }
 }
+
+/**
+ * A `swap` member function that is used as follows:
+ * ```
+ * obj1.swap(obj2)
+ * ```
+ */
+private class MemberSwap extends TaintFunction, MemberFunction {
+  MemberSwap() {
+    this.hasName("swap") and
+    this.getNumberOfParameters() = 1 and
+    this.getParameter(0).getType() instanceof ReferenceType and
```

Comment:
This line seems unnecessary given the postfix cast just below it.

Reply:
Good point - removed.